### PR TITLE
boards: posix: native_posix: Handle SDL events in zephyr thread

### PR DESCRIPTION
**Pull request by pdgendt**
This commit introduces a zephyr thread for SDL events. This allows to use zephyr mutex locking with SDL callbacks.

Relates to #50080; as this calls `gpio_emul_input_set` which locks a mutex, and asserts if called from an idle thread (which apparently is the case for the posix main thread).

### CODE REVIEW

File: boards/posix/native_posix/Kconfig [hunk 1]
```diff
@@ -26,4 +26,26 @@ config HAS_SDL
...
 	  This option specifies that the target board has SDL support
 
-endif
+if HAS_SDL
+
+config SDL_THREAD_INTERVAL
+	int "SDL thread sleep interval"
+	default 10
+	help
+	  Sleep interval time of SDL thread to handle events in microseconds.
```
Comment (aescolar) [inline boards/posix/native_posix/Kconfig:35]: Beware that k_sleep only has the resolution of the system tick. Meaning not less than 10 milliseconds by default.
=> this definition seems a bit off

File: boards/posix/native_posix/Kconfig [hunk 2]
```diff
@@ -26,4 +26,26 @@ config HAS_SDL
...
+
+config SDL_THREAD_PRIORITY
+	int "SDL thread priority"
+	default 0
+	help
+	  Priority of SDL thread to handle events.
+
+config SDL_THREAD_STACK_SIZE
+	int "SDL thread stack size"
+	default 2048
```
Comment (aescolar) [inline boards/posix/native_posix/Kconfig:45]: There is no need to define a stack size for this. You can just use ARCH_POSIX_RECOMMENDED_STACK_SIZE See https://github.com/zephyrproject-rtos/zephyr/blob/main/arch/posix/Kconfig#L12

**Review by aescolar — CHANGES_REQUESTED** [review]

**Comment by aescolar** [comment]
>(which apparently is the case for the posix main thread).

Just for info: The native_posix main thread is a separate thread than the Zephyr SW threads. If you happen to call Zephyr APIs from it, you confuse the Zephyr kernel :)

**Comment by pdgendt** [comment]
> If you call Zephyr APIs from it, you confuse the Zephyr kernel :)

That is exactly the problem :)

**Review by aescolar — COMMENTED** [review]
You can still define a kconfig option for the delay :), just not a too tiny amount

**Review by aescolar — APPROVED** [review]

**Comment by pdgendt** [comment]
@aescolar FYI when enabling `CONFIG_STACK_SENTINEL` the `ARCH_POSIX_RECOMMENDED_STACK_SIZE` should be set to 44 instead of 40 in case of 64 bit, to prevent a stack overflow. Not sure for 32 bit though.

**Comment by pdgendt** [comment]
> Would you so kind as to send a PR to fix it?

Done #50416

**Review by carlescufi — APPROVED** [review]

**Pull request merged**